lib/ReactViews/Workbench/Controls/Legend.jsx

```diff
@@ -17,6 +18,11 @@ const Legend = React.createClass({
 
     onImageError(legend) {
         legend.imageHasError = true;
+        this.forceUpdate();
+    },
+
+    componentWillMount() {
+        this.legends = this.getLegends();
```

Comment:
I think doing this on mount only will cause problems.  The list of legends may change asynchronously as the catalog item loads, and the UI needs to react to that.